lib/maze/hooks/appium_hooks.rb

```diff
@@ -83,7 +83,32 @@ def write_session_links
                   "https://automate.browserstack.com/dashboard/v2/search?query=#{build}&type=builds"
                 end
           if ENV['BUILDKITE']
-            $logger.info Maze::LogUtil.linkify url, 'BrowserStack session(s)'
+            build_info = Maze::BrowserStackUtils.build_info config.username,
+                                                            config.access_key,
+                                                            build
+
+            $logger.info Maze::LogUtil.linkify url, 'BrowserStack Build URL'
+
+            session_number = 1
+
+            build_info.each do |session|
+              $logger.info "Uploading Device Logs for Session #{session_number}"
+
+              Maze::BrowserStackUtils.download_log config.username,
+                                                   config.access_key,
+                                                   session['automation_session']['device_logs_url'],
+                                                   session_number
+
+              $logger.info "Uploading Appium Logs for Session #{session_number}"
+
+              Maze::BrowserStackUtils.download_log config.username,
+                                                   config.access_key,
+                                                   session['automation_session']['appium_logs_url'],
+                                                   session_number
```

Comment:
There's a problem in that it's trying to download the logs before the tests have been run (so it's creating empty files).  In fact, even if you do that you'll find there is a lag, as it takes time for BrowserStack to pull logs from the device.  If it's not ready, the contents of the file will be something like:
```
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message>
```

It makes me wonder how we want the feature to work - always taking time to do the download _could_ be annoying, but equally it could be useful for local running - so I don't think we should hide it inside the `if ENV['BUILDKITE']`. We could add a new command line option and default depending on whether we're running locally or on Buildkite, or perhaps we should ask our users what they prefer.